src/commonMain/kotlin/exchange.dydx.abacus/output/account/SubaccountPosition.kt

```diff
@@ -104,17 +109,26 @@ data class SubaccountPosition(
                         )
                     }
 
-                    val realizedPnlPercent = TradeStatesWithDoubleValues.create(
-                        existing?.realizedPnlPercent,
-                        parser,
-                        parser.asMap(data["realizedPnlPercent"]),
-                    )
+                    val realizedPnlPercent =
+                        if (position != null) {
+                            TradeStatesWithDoubleValues(
+                                current = position.calculated[CalculationPeriod.current]?.realizedPnlPercent,
+                                postOrder = position.calculated[CalculationPeriod.post]?.realizedPnlPercent,
+                                postAllOrders = position.calculated[CalculationPeriod.settled]?.realizedPnlPercent,
+                            )
+                        } else {
```

Comment:
Is the position null check basically the `staticTyping` flag?

Reply:
Yes, the caller will always set position = null when staticTyping is off.